db/database_test.go

```diff
@@ -3667,3 +3667,36 @@ func TestInject1xBodyProperties(t *testing.T) {
 	assert.Equal(t, "value", resBody["key"])
 	assert.True(t, resBody[BodyDeleted].(bool))
 }
+
+// TestSettingSyncInfo:
+//   - Purpose of the test is to call both SetSyncInfoMetaVersion + SetSyncInfoMetadataID in different orders
+//     asserting that the operations preserve the metadataID/metaVersion
+func TestSettingSyncInfo(t *testing.T) {
```

Comment:
I commented about this above, but I actually think that this test will probably read better if it is a parameterized test which just contains the starting state of a document.